includes/content-distribution/class-cap-authors.php

```diff
@@ -0,0 +1,166 @@
+<?php
+/**
+ * Newspack Network Co-Authors Plus authors.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack_Network\Content_Distribution;
+
+use Newspack_Network\Content_Distribution as Content_Distribution_Class;
+use Newspack_Network\Debugger;
+use Newspack_Network\User_Update_Watcher;
+use WP_Post;
+
+/**
+ * This class handles the Guest Contributors Newspack offers for CAP.
+ *
+ * For Co-Author Plus Guest Authors, see Cap_Guest_Authors.
+ */
+class Cap_Authors {
+
+	/**
+	 * Get things going.
+	 *
+	 * @return void
+	 */
+	public static function init(): void {
+		if ( ! self::is_co_authors_plus_active() ) {
+			return;
+		}
+
+		add_action( 'set_object_terms', [ __CLASS__, 'on_cap_authors_change' ], 20, 6 );
+		add_filter( 'newspack_network_multiple_authors_for_post', [ __CLASS__, 'get_outgoing_for_post' ], 10, 2 );
+
+		if ( defined( 'NEWSPACK_ENABLE_CAP_GUEST_AUTHORS' ) && NEWSPACK_ENABLE_CAP_GUEST_AUTHORS ) {
+			// Support CAP Guest Authors.
+			Cap_Guest_Authors::init();
+		}
+	}
+
+	/**
+	 * Helper to check if Co-Authors Plus is active.
+	 *
+	 * @return bool Whether Co-Authors Plus is active.
+	 */
+	public static function is_co_authors_plus_active(): bool {
+		global $coauthors_plus;
+
+		return $coauthors_plus instanceof \CoAuthors_Plus && function_exists( 'get_coauthors' );
+	}
+
+	/**
+	 * Action callback.
+	 *
+	 * Add a postmeta entry with the Co-Authors Plus authors for outgoing posts.
+	 *
+	 * @param int    $object_id The object ID.
+	 * @param array  $terms The terms.
+	 * @param array  $tt_ids The term taxonomy IDs.
+	 * @param string $taxonomy The taxonomy.
+	 * @param bool   $append Whether to append.
+	 * @param array  $old_tt_ids The old term taxonomy IDs.
+	 *
+	 * @return void
+	 */
+	public static function on_cap_authors_change( $object_id, $terms, $tt_ids, $taxonomy, $append, $old_tt_ids ): void {
+		if ( 'author' !== $taxonomy ) { // Co-Authors Plus author taxonomy.
+			return;
+		}
+		// If the terms are the same, we don't need to do anything. Note that one array has string values and one has
+		// int values, so we use array_map with the intval for the comparison.
+		if ( array_map( 'intval', $old_tt_ids ) === array_map( 'intval', $tt_ids ) ) {
+			return;
+		}
+
+		try {
+			$outgoing_post = new Outgoing_Post( $object_id );
+			if ( ! $outgoing_post->is_distributed() ) {
+				return;
+			}
+
+			Content_Distribution_Class::distribute_post_partial( $outgoing_post->get_post(), [ 'multiple_authors' ] );
```

Comment:
```suggestion
			Content_Distribution_Class::queue_post_distribution( $object_id, 'multiple_authors' );
```

Although this should work well, we can use the `queue_post_distribution()` here. If the same request runs other distribution requests for the same post, it'll run only once on shutdown.

The try/catch here is also unnecessary. If we stick with the current approach, `distribute_post_partial()` accepts a post ID integer and the `is_post_distributed()` method seems more suitable here.